packages/core/src/access.ts

```diff
@@ -90,35 +126,67 @@ export function canWriteResource(accessPolicy: AccessPolicy, resource: Resource)
  * @param resource The resource.
  * @param readonlyMode True if the resource is being read.
  * @returns True if the resource matches the access policy.
+ * @deprecated Use satisfiedAccessPolicy() instead.
  */
 export function matchesAccessPolicy(accessPolicy: AccessPolicy, resource: Resource, readonlyMode: boolean): boolean {
   if (accessPolicy.resource) {
     for (const resourcePolicy of accessPolicy.resource) {
-      if (matchesAccessPolicyResourcePolicy(resource, resourcePolicy, readonlyMode)) {
+      if (
+        matchesAccessPolicyResourcePolicy(
+          resource,
+          readonlyMode ? AccessPolicyInteraction.READ : AccessPolicyInteraction.UPDATE,
+          resourcePolicy
+        )
+      ) {
         return true;
       }
     }
   }
   return false;
 }
 
+/**
+ * Check that there is an access policy permitting the given resource interaction, returning the matching policy object.
+ * @param resource The resource being acted upon.
+ * @param interaction The interaction being performed on the resource.
+ * @param accessPolicy The relevant access policy for the current user.
+ * @returns The satisfied access policy, or undefined if the access policy does not permit the given interaction.
+ */
+export function satisfiedAccessPolicy(
```

Comment:
Similar to above, I think this should be `satisfies` rather than `satisfied`?

Reply:
Since the function returns the resource policy that was satisfied for the interaction, I changed it from `satisfies` to `satisfied` — very open to suggestions on the name though